### DISCUSSION TITLE
How to get the server id from handleError(Error error, uint32_t token)

### DISCUSSION THREAD

**Discussion by guilhermeaiolfi** [eModbus/eModbus#203] [Q&A] [answered]
Hi,

I want to control how many times one slave can return an error before disabling it. To do so, I need to get its server id in the handlerError function. But I only get the error message and token there. Is there anyway to get the server id in that function?

I looked for a way to get a request for a given token (if they are still in the queue), but I haven't found a way.

Thanks

**Comment by Miq1** [comment, marked as answer]
That is what the token is for - you are supposed to know from it which request led to the response you are receiving. So you might have to set up your own reference from token to serverIDs and whatever.

**Reply by guilhermeaiolfi** [reply]
Yeap, but we already have that internally, in the request object:

```
            instance->onError(response.getError(), request.token);
```

Couldn't be pass that request to the error so we have more information if necessary?

**Reply by Miq1** [reply]
You can use the ``onResponse`` handler instead and sort out error responses yourself.

**Reply by guilhermeaiolfi** [reply]
Perfect. Thank you so much.
